packages/chain/consensus/action.go

```diff
@@ -717,18 +718,34 @@ func (c *consensus) setNewState(msg *messages.StateTransitionMsg) bool {
 		return false
 	}
 
-	c.stateOutput = msg.StateOutput
-	c.currentState = msg.State
+	// If c.stateOutput.GetStateIndex() == msg.StateOutput.GetStateIndex() and the new state output is not a governance update, then either,
+	// a) c.stateOutput is the same as msg.StateOutput and there is no need to reasign c.stateOutput or b) msg.StateOutput is a regular state update
+	// output and c.stateOutput is a governance update output with the same index; in such case governance update is the last and should be taken
+	// into account. Regular state output is overwriden by governance uptade output and should be ignored.
+	// TODO: it is assumed, that at most one governance update may occur in between regural state updates. The situation of several consecutive
+	// governance updates must be yet to be discussed, designed and implemented. The main problem is that there is no way in knowing the exact order
```

Comment:
```suggestion
	// governance updates are yet to be discussed, designed and implemented. The main problem is that there is no way in knowing the exact order
```
what does this mean?

Reply:
This means that if two different governance rotation transactions are approved one after another without state transaction in between, there is no guarantee that the nodes are going to be synchronised correctly. Maybe the wording is difficult to understand. I'll try to improve it. And thanks for all your spelling corrections. I am terrible at that.